### ISSUE TITLE
[YSQL] Failing test YbAdminSnapshotScheduleUpgradeTestWithYsql.PgsqlTestMigrationFromEarliestSysCatalogSnapshot

### ISSUE THREAD

**Issue by NatashaSerebryanaya** [yugabyte/yugabyte-db#15896]
Jira Link: [DB-5296](https://yugabyte.atlassian.net/browse/DB-5296)
### Description

YbAdminSnapshotScheduleUpgradeTestWithYsql.PgsqlTestMigrationFromEarliestSysCatalogSnapshot started fail quite regularly (but not always) in release and fastdebug builds: https://detective.dev.yugabyte.com/stability/test?analyze_trends=true&class=YbAdminSnapshotScheduleUpgradeTestWithYsql&name=PgsqlTestMigrationFromEarliestSysCatalogSnapshot

The error message:
```
../../src/yb/tools/yb-admin-snapshot-schedule-test.cc:2590
Failed
Bad status: Runtime error (yb/util/subprocess.cc:615): Subprocess '/nfusr/alma8-gcp-cloud/jenkins-worker-t2n37y/jenkins/jenkins-github-yugabyte-db-alma8-master-clang15-release-609/yugabyte-db/build/release-clang15-linuxbrew-full-lto-ninja/tests-tools/../bin/yb-admin' terminated with non-zero exit status 256: Error running upgrade_ysql: Network error (yb/yql/pgwrapper/libpq_utils.cc:329): Unable to upgrade YSQL cluster: Failed to apply migration 'V31__10595__pg_stat_progress_create_index.sql' to a database template1: Execute failed: 7, message: ERROR:  column s.param16 does not exist
LINE 29:         S.param16 AS blocks_total,
                 ^ (pgsql error 42703) (aux msg ERROR:  column s.param16 does not exist
LINE 29:         S.param16 AS blocks_total,
                 ^)
```
Reproducible locally - 9 fails out of 10 runs with the release build
```
./yb_build.sh release --lto=full --cxx-test tools_yb-admin-snapshot-schedule-test --gtest_filter YbAdminSnapshotScheduleUpgradeTestWithYsql.PgsqlTestMigrationFromEarliestSysCatalogSnapshot
```

[DB-5296]: https://yugabyte.atlassian.net/browse/DB-5296?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

**Comment by NatashaSerebryanaya** [comment]
Didn't notice https://github.com/yugabyte/yugabyte-db/issues/15873 before.
I assume it's the same issue just another test.

**Comment by fizaaluthra** [comment]
Should be fixed as of [00e886216e63118f24e15b4d16527354f23f5359](https://github.com/yugabyte/yugabyte-db/commit/00e886216e63118f24e15b4d16527354f23f5359)
